File: ios/RNSha256Module.swift
import ExpoModulesCore
import CryptoKit

extension Digest {
    var bytes: [UInt8] { Array(makeIterator()) }
    var data: Data { Data(bytes) }

    var hexStr: String {
        bytes.map { String(format: "%02X", $0) }.joined()
    }
}

public class RNSha256Module: Module {
    public func definition() -> ModuleDefinition {
        Name("RNSha256")
        
        Function("md5") { (input: String) -> String in
            guard let data = input.data(using: .utf8) else { return "" }
            let digest = Insecure.MD5.hash(data: data)
            return digest.hexStr
        }
        
        Function("sha1") { (input: String) -> String in
            guard let data = input.data(using: .utf8) else { return "" }
            let digest = Insecure.SHA1.hash(data: data)
            return digest.hexStr
        }
        
        Function("sha256") { (input: String) -> String in
            guard let data = input.data(using: .utf8) else { return "" }
            let digest = SHA256.hash(data: data)
            return digest.hexStr
        }
        
        Function("sha384") { (input: String) -> String in
            guard let data = input.data(using: .utf8) else { return "" }
            let digest = SHA384.hash(data: data)
            return digest.hexStr
        }
        
        Function("sha512") { (input: String) -> String in
            guard let data = input.data(using: .utf8) else { return "" }
            let digest = SHA512.hash(data: data)
            return digest.hexStr
        }
    }
}

File: android/src/main/java/com/nasirimehr/sha256/RNSha256Module.kt
package com.nasirimehr.sha256

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import java.security.MessageDigest

class RNSha256Module : Module() {
  private fun hashString(type: String, input: String): String {
    val HEX_CHARS = "0123456789ABCDEF"
    val bytes = MessageDigest
      .getInstance(type)
      .digest(input.toByteArray())
    val result = StringBuilder(bytes.size * 2)

    bytes.forEach {
      val i = it.toInt()
      result.append(HEX_CHARS[i shr 4 and 0x0f])
      result.append(HEX_CHARS[i and 0x0f])
    }

    return result.toString()
  }
  override fun definition() = ModuleDefinition {
    Name("RNSha256")

    Function("md5") {input:String ->
      return@Function hashString("MD5",input)
    }

    Function("sha1") {input:String ->
      return@Function hashString("SHA-1",input)
    }

    Function("sha256") {input:String ->
      return@Function hashString("SHA-256",input)
    }

    Function("sha384") {input:String ->
      return@Function hashString("SHA-384",input)
    }

    Function("sha512") {input:String ->
      return@Function hashString("SHA-512",input)
    }
  }
}
